[joelgoh1] iP

joelgoh1 opened the pull request:
# Blawg 
> ":D" - me 

Blawg 🐱 frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/joelgoh1/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😆 

And it is **FREE**!

Features:

- [x] Managing tasks
-  [ ] Managing deadlines (coming soon)
-  [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```java
public static void main(String[] args) {
        new Blawg().run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

joelgoh1 committed:
Rename Duke.java to Blawg.java, add greet and exit functionality

joelgoh1 committed:
add echo functionality to Blawg.java

joelgoh1 committed:
add add and list functionality to Blawg.java

joelgoh1 committed:
add Mark as Done functionality to Blawg.java, add Task.java class

joelgoh1 committed:
add Todos, events, deadlines functionalities

joelgoh1 committed:
Automated Text UI testing

joelgoh1 committed:
Add Error handling

joelgoh1 committed:
Add delete functionality

joelgoh1 committed:
Add storage functionalities with storage class

joelgoh1 committed:
Storage: add ability to store task to write and read from

Users can have their data automatically stored after each use.
Past data from previous usage can be retrieved so that users do not need to retype tasks.

Let's,
* add storage functionality

joelgoh1 committed:
Add basic date parsing ability

joelgoh1 committed:
Addcommand: basic date and time parsing functionality

Chatbot accepting only valid dates is more user-friendly because users
cannot accidentally type and invalid date

Let's,
* add basic date and time parsing to AddCommand

joelgoh1 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

joelgoh1 committed:
Add gradle support as well as JUnit Testing for Storage and Parser class

joelgoh1 committed:
Update build.gradle shadowJar to create Jar file for project

ashleygoh1 commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,95 @@
+    }
+
+    public ArrayList<Task> read() throws FileError, FileAccessError {
+        ArrayList<String> content = new ArrayList<>();
```
contents (plural form)
since the array will store many contents

ashleygoh1 reviewed (commented):
LGTM! amazing code.

joelgoh1 committed:
Add JavaDoc to classes and method to generate explanatory tooltips

joelgoh1 committed:
Update class name to better represent command

joelgoh1 committed:
Edit certain portions so that all comply with Java coding standards

joelgoh1 committed:
Edit certain portions so that all comply with Java coding standards

joelgoh1 committed:
Add find functionalities, allowing users to find tasks containing words they wish to search for

joelgoh1 committed:
fix conflicts in merging

joelgoh1 committed:
fix conflicts in merging to branch-Level-9

joelgoh1 committed:
Add checkstyle files and change code to align with checkstyle requirements

Arixeyeion commented on src/main/java/Blawg.java:
```diff
@@ -0,0 +1,58 @@
+import commands.Command;
+import exceptions.BlawgException;
+import parser.Parser;
+import storage.Storage;
+import tasklist.TaskList;
+import ui.Ui;
```
I like that everything is split up into packages.

Arixeyeion commented on src/main/java/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,29 @@
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws FileError {
+        Task removedTask = tasks.getTaskList().get(index - 1);
+        tasks.getTaskList().remove(index - 1);
+        String result = "Noted. I've removed this task:\n"
+                + removedTask
+                + String.format("Now you have %d tasks in the list\n", tasks.getTaskList().size());
```
For this portion, I would suggest starting the line with String.format. There is no need to + removedTask as that already uses the toString method. Instead doing String.format("Noted...\n%s...", removedTask.toString()...) would allow you to view the entire string as if you were reading a whole sentence or paragraph.

Arixeyeion reviewed (commented):
Overall, I like how everything is compartmentalized, this makes everything easier to understand as each class performs a "single" task. I also enjoyed the personalization of the messages the bot produced. Just some nits about the usage of String.format().

joelgoh1 committed:
Add GUI capabilities with JavaFX and FXML

joelgoh1 committed:
Add assertions to getResponse methods to ensure
that input is not empty

joelgoh1 committed:
Parser class: parses userInput to Commands

Parse method is long and does not comply with
SLAP.

Create more methods to handle each specific
user interaction improves readability.

Let's create separate methods for each user
interaction.

joelgoh1 committed:
Merge pull request #3 from joelgoh1/A-Assertions

Add assertions to getResponse method

joelgoh1 committed:
Merge pull request #2 from joelgoh1/A-CodeQuality

Parser class: parses userInput to Commands

joelgoh1 committed:
Add help functionalities

joelgoh1 committed:
Update readme.md

joelgoh1 committed:
Prepare product to be released

joelgoh1 committed:
Minor bug fixes
